# 数组相关的面试题

## 1. 如何跳出数组的forEach的循环

```js
Array.prototype.myForEach = function(callback,self){
    let _arr = [...this];
    for(let i = 0;i<_arr.length;i++){
        callback.call(self,_arr[i],i,_arr)
    }
}
```
它本身也是借用的 for 循环 来处理当前元素，执行回调方法，在这个方法里肯定是跳不出来的。所以采用抛异常的方式，使用try catch块进行包裹


```js
let arr = [...new Array(10).keys()];
try{
    arr.forEach(item=>{
        console.log(`item:${item}`)
        if(item>5) throw new Error("break");
    })
}catch(err){
    if(err.message === "break")	
        console.log("break success!")
    else 
        console.error(err)
}
```